.github/workflows/test-canvas.yml

```diff
@@ -5,61 +5,61 @@
 
 name: Canvas Extractor - Test
 on:
-  pull_request:
-    paths:
-      - 'src/canvas-extractor/**'
-      - '!src/canvas-extractor/**/*.md'
-  push:
-    paths:
-      - 'src/canvas-extractor/**'
-      - '!src/canvas-extractor/**/*.md'
-    branches:
-      - main
-  workflow_dispatch:
+    pull_request:
+        paths:
+            - "src/canvas-extractor/**"
+            - "!src/canvas-extractor/**/*.md"
+    push:
+        paths:
+            - "src/canvas-extractor/**"
+            - "!src/canvas-extractor/**/*.md"
+        branches:
+            - main
+    workflow_dispatch:
 
 jobs:
-  test-canvas-extractor:
-    name: Run unit, style, and type checks
-    runs-on: ubuntu-20.04
-    env:
-        PACKAGE_NAME: canvas-extractor
-        SRC_DIR: src/canvas-extractor
-        LOCK_FILE: src/canvas-extractor/poetry.lock
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+    test-canvas-extractor:
+        name: Run unit, style, and type checks
+        runs-on: ubuntu-20.04
+        env:
+            PACKAGE_NAME: canvas-extractor
+            SRC_DIR: src/canvas-extractor
+            LOCK_FILE: src/canvas-extractor/poetry.lock
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
 
-      - name: Install Python 3.9
-        uses: actions/setup-python@3105fb18c05ddd93efea5f9e0bef7a03a6e9e7df
-        with:
-          # This one needs to be hard-coded to avoid 3.9.5, due to a bug
-          # in one of the dependencies
-          python-version: '3.9.4'
+            - name: Install Python 3.9
+              uses: actions/setup-python@3105fb18c05ddd93efea5f9e0bef7a03a6e9e7df
+              with:
+                  # This one needs to be hard-coded to avoid 3.9.5, due to a bug
+                  # in one of the dependencies
+                  python-version: "3.9.4"
```

Comment:
If you move "Install Python" after "Install Poetry", then you can setup Poetry caching on the "Install Python" step and then remove the "get virtual env path" and "setup dependency caching" steps.